stdlib/public/Darwin/Accelerate/vDSP_FIR.swift

```diff
@@ -0,0 +1,85 @@
+import Accelerate
+
+extension vDSP {
+    
+    /// FIR filtering with decimation and antialiasing; single-precision.
+    ///
+    /// - Parameter source: Single-precision input vector.
+    /// - Parameter decimationFaction: The integer factor by which to divide the sampling rate.
+    /// - Parameter filter: Filter to use during the downsampling operation.
+    /// - Parameter result: Single-precision output vector.
+    @inline(__always)
+    @available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+    public static func downsample<T, U, V>(_ source: U,
+                                    by decimationFaction: Int,
+                                    using filter: T,
+                                    result: inout V)
```

Comment:
The parameter naming here seems suboptimal. We usually want labels to say what the parameter is, but `by` and `using` are just prepositions. It's unclear what's going on here:
```
downsample(a, by: 4, using: c, result: b)
```
I would prefer to see something a bit more explicit, like:
```
downsample(a, decimationFactor: 4, filter: c, result: b)
```

Reply:
changed (pending push)